hs_composite_resource/models.py

```diff
@@ -194,65 +163,146 @@ def get_metadata_xml(self, pretty_print=True, include_format_elements=True):
 
         return etree.tostring(RDF_ROOT, pretty_print=pretty_print)
 
-    def create_aggregation_xml_documents(self, aggregation_name=None):
+    def _recreate_fileset_xml_docs(self, folder):
+        """Recreates xml files for all fileset aggregations that exist under the path 'folder'
+        as well as for any parent fileset that may exist relative to path 'folder'
+        """
+
+        filesets = self.filesetlogicalfile_set.filter(folder__startswith=folder)
+        for fs in filesets:
+            fs.create_aggregation_xml_documents()
+
+        # Also need to recreate xml doc for any parent fileset that may exist relative to path
+        # *folder*
+        if '/' in folder:
+            path = os.path.dirname(folder)
+            parent_fs = self.get_fileset_aggregation_in_path(path)
+            if parent_fs is not None:
+                parent_fs.create_aggregation_xml_documents()
```

Comment:
Wouldn't ```parent_fs.create_aggregation_xml_documents()``` create aggregation xml files for all its children including all filesets under the path folder? If so, lines 171-173 probably should go to the ```else``` phrase meaning they are only needed when there were no parent fileset.

Reply:
```parent_fs.create_aggregation_xml_documents()``` won't create xml files for any child aggregation.